src/views/withNavigationFocus.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import hoistStatics from 'hoist-non-react-statics';
+
+export default function withNavigationFocus(Component) {
+  class ComponentWithNavigationFocus extends React.Component {
+    static displayName = `withNavigationFocus(${Component.displayName ||
+      Component.name})`;
+
+    state = {
+      isFocused: false,
+    };
+
+    componentDidMount() {
+      if (this.props.navigation) {
```

Comment:
I think it would be great if we could use the navigtion from context if the user forgets to pass it in as a prop. Take a look at the other HOC, `withNavigation`.